aiida/backends/sqlalchemy/models/log.py

```diff
@@ -29,36 +29,48 @@ class DbLog(Base):
 
     id = Column(Integer, primary_key=True)
 
-    uuid = Column(UUID(as_uuid=True), default=uuid_func)
+    uuid = Column(UUID(as_uuid=True), default=uuid_func, unique=True)
 
     time = Column(DateTime(timezone=True), default=timezone.now)
     loggername = Column(String(255), index=True)
     levelname = Column(String(255), index=True)
 
-    objname = Column(String(255), index=True)
-    objuuid = Column(UUID(as_uuid=True), default=uuid_func, unique=False, index=True)
+    dbnode_id = Column(
+        Integer,
+        ForeignKey('db_dbnode.id', deferrable=True, initially="DEFERRED", ondelete="CASCADE"),
+        nullable=False
+    )
 
     message = Column(Text(), nullable=True)
     _metadata = Column('metadata', JSONB)
 
-    def __init__(self, time, uuid=None, loggername="", levelname="", objname="", objuuid=None,
-                 message=None, metadata=None):
+    ######### RELATIONSSHIP #########
+    dbnode = relationship(
+        'DbNode',
+        backref=backref('dblogs', passive_deletes='all', cascade='merge')
+    )
+
+    def __init__(self, time, uuid=None, loggername="", levelname="", dbnode_id=None,
```

Comment:
`dbnode_id` should not be empty either correct? And if `loggername` and `levelname` *have* to be specified, better to not define an empty string as default.

Reply:
I agree about dbnode_id. I am not so sure about `loggername` and `levelname`. It depends how you use it. You know better than me the use cases at the workflows so, I agree with you